src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -118,22 +118,20 @@ bfsForestFrom' vs g = evalState (explore vs) IntSet.empty where
 -- @
 -- bfs vs 'empty'                                         == []
 -- bfs [] g                                             == []
--- bfs [1] ('edge' 1 1)                                   == [[[1]]]
--- bfs [1] ('edge' 1 2)                                   == [[[1],[2]]]
--- bfs [2] ('edge' 1 2)                                   == [[[2]]]
+-- bfs [1] ('edge' 1 1)                                   == [[1]]
+-- bfs [1] ('edge' 1 2)                                   == [[1],[2]]
+-- bfs [2] ('edge' 1 2)                                   == [[2]]
 -- bfs [3] ('edge' 1 2)                                   == []
--- bfs [1,2] ('edge' 1 2)                                 == [[[1],[2]]]
--- bfs [2,1] ('edge' 1 2)                                 == [[2]],[[1]]
+-- bfs [1,2] ('edge' 1 2)                                 == [[1],[2]]
```

Comment:
I'm confused by this example: shouldn't both vertices be in the first group since they are reachable in 0 steps?

Reply:
I got confused by this earlier as well. Basically, the tree from 1 is fully expanded before considering 2.

I remembered when first adding `bfsForestFrom`, I got tripped up by this too, since if the input vertices get searched "in parallel" 
```
bfsForest g = bfsForestFrom (vertexList g)
```
returns a forest of singleton trees for each vertex. This approach was to mirror the dfs API, though I'm not sure it's the most natural approach.

Reply:
I think it would be great to have the property that vertices in k'th group are k steps away from the source vertices. BFS is primarily used when the distance information is important and I think having such API would be more useful than the one mirroring the approach used in DFS.

Presumably, it's not very difficult implementation-wise?

Reply:
Yeah that makes sense. I think it might actually be an easier implementation, since I had to be careful about the evaluation order with the current implementation. Should I go ahead and make the changes in this pull request?

Reply:
Great! Sure, feel free to do the changes right here.

Reply:
I think this example is still incorrect but the test does seem to use `[[1,2]]` already.